Added utilities to instrument kernel bandwidth numbers

Chillee opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #95355



cc @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

Looks like

![image](https://user-images.githubusercontent.com/6355099/221048077-33aeff50-0951-42c9-89e9-22049db4f94d.png)

Chillee committed:
Added utilities to instrument kernel bandwidth numbers

[ghstack-poisoned]

ngimel reviewed (approved):
Add a config option for this?

jansel commented on torch/_inductor/triton_ops/autotune.py:
```diff
@@ -273,6 +317,14 @@ def save_cache_hook(cfg):
     mutated_arg_names = meta.pop("mutated_arg_names", ())
 
     def decorator(fn):
+        # Uncomment this to profile bandwidth for each Triton kernel
```
Perhaps add a config flag?

Chillee replied:
Yeah I'll add an env variable.

Chillee commented on torch/_inductor/triton_ops/autotune.py:
```diff
@@ -204,6 +204,50 @@ def run(self, *args, grid, stream):
+
+
+class DebugAutotuner(CachingAutotuner):
+    def __init__(self, *args, regex_filter="", **kwargs):
```
Oh I also wanted to add a way to filter the triton kernels so not all of them would get run.

Chillee commented on torch/_inductor/triton_ops/autotune.py:
```diff
@@ -204,6 +204,50 @@ def run(self, *args, grid, stream):
+        if isinstance(referrer, dict):
+            for k, v in referrer.items():
+                if v is obj:
+                    obj_names.append(k)
```
Well, we could pass it down in the codegen to the CachedAutotuner object, but this was a less intrusive way of doing the same thing.

Since it's debug/instrumentation code, I didn't feel like it was necessarily worth doing the modification.

Chillee committed:
Update on "Added utilities to instrument kernel bandwidth numbers"




cc soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

Looks like

<img width="444" alt="image" src="https://user-images.githubusercontent.com/6355099/220815020-c62128d9-9b12-4d8b-ab54-219f92b5d5b0.png">


[ghstack-poisoned]

Chillee committed:
Update on "Added utilities to instrument kernel bandwidth numbers"




cc soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

Looks like

![image](https://user-images.githubusercontent.com/6355099/221048077-33aeff50-0951-42c9-89e9-22049db4f94d.png)


[ghstack-poisoned]

Chillee commented:
@pytorchbot merge
